### ISSUE TITLE
Studio Fade Out Generates separate clips and fails when used at intermediate points due to insufficient space to paste

### ISSUE THREAD

**Issue by Charlie-Ramirez-Animation-Studios-de-MX** [audacity/audacity#6142]
### Bug description

Like the title, using the Studio Fade out effect on a 48KHz Stereo Track tends to generate separate clips over the region to which the effect is applied occasionally and fails when used at intermediate points in the track [anything other than the final portion with empty field] showing an error due to insufficient space to paste on the track
<img width=500 src=https://github.com/audacity/audacity/assets/64709446/fe4bf8d2-25a8-4414-9eca-57418fbc8aba>
<img width=500 src=https://github.com/audacity/audacity/assets/64709446/76319d93-916d-4f93-ada0-c93437238c62>

### Steps to reproduce

1. Start any project, preferably with a stereo track that is at least 3 minutes long [preferably Opus File at 48KHz Sample rate, _although there does not seem to be a specific track 48/44.1 distinction_]. 
2. Select the Studio Fade out Effect and try making Fades of various lengths at various points in the track. **Eventually some of the attempts will generate a Separate track** (commonly when applied at the end of the track) or show the effect of Insufficient space to insert (this in intermediate points or without white space)

### Expected behavior

The effect should not cut or generate separate clips over the manipulated portion.
If applied at intermediate points, it should not trigger errors due to lack of space or depend on the permission to insert/move track clips.

### Actual behavior

Studio Fade Out works but generates clips that must be selected and joined manually. Cannot be used at points other than the Strict End of the track due to "insufficient space to insert"

The behavior is somewhat random. It doesn't always happen the first time. and appears to be dependent on the selected length of the effect (very short durations of <2 seconds do not seem to trigger it)

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

### Additional context

Running on a Windows 11 Pro 23H2 system
With .OPUS tracks imported via FFMPEG filter.
Applying an Effects Macro in order: _[Probably unrelated]_
-Entire track selection
-Normalization to -1dB eliminating DC in independent channels
-10Hz 48dB high pass filter
-Channel independent volume normalization to -14LUFS

**Comment by petersampsonaudacity** [comment]
This works for me with 3.4.2 with no split.

But with the latest 3.5.0 master and the latest 3.5.0 Beta when using Studio Fade Out I do get the split.

_@SteveDaulton can you shed any light on this please ?_

**Comment by petersampsonaudacity** [comment]
> and fails when used at intermediate points in the track [anything other than the final portion with empty field] showing an error due to insufficient space to paste on the track

This is an important use case for folk who are transcribing LPs and tapes of albums.

You normally have one or both sides with multiple songs on the single track and you will want to apply Studio Fade Out at the end of each song (i.e. at intermediate ranges in the track).

**Comment by SteveDaulton** [comment]
> But with the latest 3.5.0 master and the latest 3.5.0 Beta when using Studio Fade Out I do get the split.
> 
> @SteveDaulton can you shed any light on this please ?

Testing on Linux, most of the time the effect is applied and no split is created, but occasionally a split is created. On those occasions when the split is created, the processed audio is slightly shorter than the audio that was selected. In the case that I am looking at now, the processed audio has shrunk by 2 samples.

It used to be standard behaviour  for a split line to be created when the processed audio was a different length from the original audio. I don't know what the intended behaviour is these days, it appears to be undefined. Some generators (example "Chirp") allow you generate a different length and no split lines are created, whereas other generators (example "Pluck") fail ("not enough room" error) if the selection length is not set to exactly match the selection (a different bug?)

It is not normal (a bug) for the processed audio from "Studio Fade Out" to be a different length from the original selected audio. The processed audio is supposed to be exactly the same  length as the original selected audio.

In other words the "generates separate clips" part of this bug is still present.

**Comment by SteveDaulton** [comment]
A bit more testing, and the "not enough room" error still occurs sometimes. This bug is not yet fixed.

**Comment by Charlie-Ramirez-Animation-Studios-de-MX** [comment]
2024-04-24 - Tested on new Audacity 3.5.0 Stable on Windows 11

Error still occurs 

> Testing on W10
> 
> This works for me with 3.4.2 with no split.
> 
> But with the latest 3.5.0 master and the latest 3.5.0 Beta when using Studio Fade Out I do get the split.
> 
> _@SteveDaulton can you shed any light on this please ?_
> 
> Curiously if you make a selection in the middle (or beginning) of a clip you do _**NOT**_ get the additional clip made when you use Studio Fade Out 

![image](https://github.com/audacity/audacity/assets/64709446/91b27e64-6c53-4677-873d-3572cf9464a3)

**Comment by LWinterberg** [comment]
Hello! We just [switched development to Audacity 4](https://forum.audacityteam.org/t/master-has-been-switched-to-4-0-0-development/120444). We've redone, or plan on redoing, various things from scratch, including the entire UI, and the part this bug is about. As such, I'm closing this bug as it will be fixed there and we won't be fixing it for the final releases of Audacity 3.
